src/DeepLinkModule/javasource/deeplink/actions/StartDeeplinkJava.java

```diff
@@ -215,10 +215,13 @@ else if (!deeplink.getAllowGuests().booleanValue() && session.getUser().isAnonym
 			//we have a valid session, further security is enforced by the application
 			
 			//first, remove old pending links
-			List<IMendixObject> pendinglinks = Core.retrieveXPathQueryEscaped(context, "//%s[%s='%s' and %s='%s']",
+			List<IMendixObject> pendinglinks = Core.retrieveXPathQuery(context, String.format("//%s[%s='%s' and %s='%s']",

```

Comment:
Now it seems we made this xpath query vulnerable to xpath injection - official deprecation recommends using `https://apidocs.mendix.com/7/runtime/com/mendix/core/Core.html#createXPathQuery-java.lang.String-` instead. (Many places)